pandas/tests/io/test_parquet.py

```diff
@@ -110,48 +110,72 @@ def df_full():
                                pd.Timestamp('20130103')]})
 
 
-def test_invalid_engine(df_compat):
+def check_round_trip(df, engine=None, path=None,
+                     write_kwargs=None, read_kwargs=None,
+                     expected=None, check_names=True):
 
-    with pytest.raises(ValueError):
-        df_compat.to_parquet('foo', 'bar')
+    if write_kwargs is None:
+        write_kwargs = {'compression': None}
 
+    if read_kwargs is None:
+        read_kwargs = {}
 
-def test_options_py(df_compat, pa):
-    # use the set option
+    if engine:
+        write_kwargs['engine'] = engine
+        read_kwargs['engine'] = engine
 
-    df = df_compat
-    with tm.ensure_clean() as path:
+    if expected is None:
+        expected = df
 
-        with pd.option_context('io.parquet.engine', 'pyarrow'):
-            df.to_parquet(path)
+    if path is None:
+        with tm.ensure_clean() as path:
+            df.to_parquet(path, **write_kwargs)
```

Comment:
can you make both of these into a loop with a kwarg in the signature ``repeat=2``

```
for _ in range(repeat):
    .....
```

Reply:
Thanks... 

Python is not native for me, I was looking for a nicer way to do that 👍